src/commands/deploy.rs

```diff
@@ -155,12 +158,57 @@ fn run_eif(
     debug!(stdout=?run_out);
 
     if !run_out.status.success() {
-        Err(failure::err_msg(format!(
+        return Err(failure::err_msg(format!(
             "failed to run enclave{:?}",
             run_out
-        )))
-    } else {
-        Ok(run_out)
+        )));
+    }
+
+    match check_enclave_status(ssh_key, url) {
+        Ok(_) => info!("Enclave up and running!"),
+        Err(err) => error!("Error: something went wrong with deployment. {}", err),
+    }
+
+    Ok(run_out)
+}
+
+fn check_enclave_status(ssh_key: &str, url: &str) -> Result<(), Error> {
+    info!("Check enclave status...");
+    let describe_out = Command::new("ssh")
+        .args([
+            "-i",
+            ssh_key,
+            format!("ec2-user@{}", url).as_str(),
+            "nitro-cli",
+            "describe-enclaves",
+        ])
+        .output()?;
+    debug!(stdout=?describe_out);
+
+    if !describe_out.status.success() {
+        return Err(failure::err_msg(format!(
+            "failed to get enclave info{:?}",
+            describe_out
+        )));
+    };
+
+    let json: Value = match from_slice(&describe_out.stdout) {
+        Ok(json) => json,
+        Err(_) => return Err(failure::err_msg("Could not parse AWS response.")),
+    };
+
+    let description = match json.as_array() {
+        Some(enclaves) => match enclaves.get(0) {
+            Some(enclave) => enclave,
+            None => return Err(failure::err_msg("Enclave not created.")),
+        },
+        None => return Err(failure::err_msg("Enclave not created.")),
+    };
+
+    match description.get("State") {
+        Some(x) if x.eq(&json!("RUNNING")) => Ok(()),
+        Some(x) => Err(failure::err_msg(format!("Enclave created, but is {}.", x))),
```

Comment:
Do you have any sense of what could potentially be returned here?

Reply:
If you do maybe for a future TODO we could parse this part too and potentially give some more info.

Reply:
According to the [doc](https://docs.aws.amazon.com/enclaves/latest/user/cmd-nitro-describe-enclaves.html), it's either "running" or "terminating", so not too many cases to handle. I can add a comment though!